scIB/metrics/ari.py

```diff
@@ -14,20 +16,31 @@ def ari(adata, group1, group2):
     """
 
     checkAdata(adata)
+    checkBatch(group1, adata.obs)
+    checkBatch(group2, adata.obs)
 
-    if isinstance(group1, str):
-        checkBatch(group1, adata.obs)
-        group1 = adata.obs[group1].tolist()
-    elif isinstance(group1, pd.Series):
-        group1 = group1.tolist()
-
-    if isinstance(group2, str):
-        checkBatch(group2, adata.obs)
-        group2 = adata.obs[group2].tolist()
-    elif isinstance(group2, pd.Series):
-        group2 = group2.tolist()
+    group1 = adata.obs[group1].to_numpy()
+    group2 = adata.obs[group2].to_numpy()
```

Comment:
What happened to the `checkBatch` calls? Also is this no longer able to take a `pd.Series`? The expected input type should probably be in the documentation above.

Reply:
`checkBatch` is called no matter the input type and I wouldn't support pd.Series, as we never use it and it wouldn't make much sense if we also provide an anndata as input. For pd.Series input, we should have a different function signature.

Reply:
makes sense! With the documentation it's all good :).